dump_utils.cpp

```diff
@@ -147,5 +147,79 @@ bool isHostQuiesced()
     }
     return false;
 }
+
+// mapping of severity enum to severity interface
+static std::unordered_map<PelSeverity, std::string> sevMap = {
+    {PelSeverity::INFORMATIONAL,
+     "xyz.openbmc_project.Logging.Entry.Level.Informational"},
+    {PelSeverity::DEBUG, "xyz.openbmc_project.Logging.Entry.Level.Debug"},
+    {PelSeverity::NOTICE, "xyz.openbmc_project.Logging.Entry.Level.Notice"},
+    {PelSeverity::WARNING, "xyz.openbmc_project.Logging.Entry.Level.Warning"},
+    {PelSeverity::CRITICAL, "xyz.openbmc_project.Logging.Entry.Level.Critical"},
+    {PelSeverity::EMERGENCY,
+     "xyz.openbmc_project.Logging.Entry.Level.Emergency"},
+    {PelSeverity::ERROR, "xyz.openbmc_project.Logging.Entry.Level.Error"},
+    {PelSeverity::ALERT, "xyz.openbmc_project.Logging.Entry.Level.Alert"}};
+
+void createPEL(
+    const std::unordered_map<std::string, std::string>& additionalData,
+    const PelSeverity& sev, const std::string& errIntf)
+{
+    try
+    {
+        auto bus = sdbusplus::bus::new_default();
+        constexpr auto loggerObjectPath = "/xyz/openbmc_project/logging";
+        constexpr auto loggerCreateInterface =
+            "xyz.openbmc_project.Logging.Create";
+        constexpr auto loggerService = "xyz.openbmc_project.Logging";
+        std::string pelSeverity =
+            "xyz.openbmc_project.Logging.Entry.Level.Informational";
+        auto itr = sevMap.find(sev);
+        if (itr != sevMap.end())
+            pelSeverity = itr->second;
+
+        sd_bus* pSD_Bus = nullptr;
+        ;
+        sd_bus_default(&pSD_Bus);
```

Comment:
get it from caller

Reply:
I don't see the caller in resource_dump_entry.cpp and system_dump_entry.cpp has this sd_bus object to be passed on. Please correct me if I am wrong with my analysis.
`void Entry::delete_()
{
    auto dumpId = id;
    // Skip the system dump delete if the dump is in progress
    // and in memory preserving reboot path
    if ((openpower::dump::util::isInMpReboot()) &&
        (status() == phosphor::dump::OperationStatus::InProgress))
    {
        log<level::INFO>(
            fmt::format(
                "Skip deleting system dump delete id({}) durng mp reboot",
                dumpId)
                .c_str());
        return;
    }

    auto srcDumpID = sourceDumpId();

    if ((!offloadUri().empty()) && (phosphor::dump::isHostRunning()))
    {
        log<level::ERR>(
            fmt::format("Dump offload in progress id({}) srcdumpid({})", dumpId,
                        srcDumpID)
                .c_str());
        elog<sdbusplus::xyz::openbmc_project::Common::Error::Unavailable>();
    }

    log<level::INFO>(fmt::format("System dump delete id({}) srcdumpid({})",
                                 dumpId, srcDumpID)
                         .c_str());

    auto path =
        std::filesystem::path(SYSTEM_DUMP_SERIAL_PATH) / std::to_string(dumpId);

    // Remove host system dump when host is up by using source dump id
    // which is present in system dump entry dbus object as a property.
    if ((phosphor::dump::isHostRunning()) && (srcDumpID != INVALID_SOURCE_ID))
    {
        try
        {
            phosphor::dump::host::requestDelete(srcDumpID,
                                                TRANSPORT_DUMP_TYPE_IDENTIFIER);
        }
        catch (const std::exception& e)
        {
            log<level::ERR>(fmt::format("Error deleteing dump from host id({}) "
                                        "host id({}) error({})",
                                        dumpId, srcDumpID, e.what())
                                .c_str());
            elog<sdbusplus::xyz::openbmc_project::Common::Error::Unavailable>();
        }
    }

    // Remove Dump entry D-bus object
    phosphor::dump::Entry::delete_();
    try
    {
        std::filesystem::remove_all(path);
    }
    catch (const std::filesystem::filesystem_error& e)
    {
        // Log Error message and continue
        log<level::ERR>(
            fmt::format("Failed to delete dump file({}), errormsg({})",
                        path.string().c_str(), e.what())
                .c_str());
    }

    log<level::INFO>(
        fmt::format("System dump entry with id({}) is deleted", dumpId)
            .c_str());

    // Log PEL for dump delete/offload
    log<level::INFO>("Log PEL for dump delete or offload");

    std::unordered_map<std::string, std::string> additionalData = {
        {"Dump ID", std::to_string(dumpId)},
        {"Filename", path},
        {"Dump Type", "System Dump"}};
    phosphor::dump::createPEL(additionalData,
                              phosphor::dump::PelSeverity::INFORMATIONAL,
                              "xyz.openbmc_project.Dump.Error.Invalidate");
}`

Reply:
<img width="1225" alt="Screenshot 2023-01-19 at 5 09 00 PM" src="https://user-images.githubusercontent.com/112170550/213433183-c376dcfe-3cb9-40af-8c06-1edd00ea84c1.png">
Getting compilation error. The sdbusplus wrapper has no direct access to it's internal sd_bus* object and async method is not yet available in the wrapper class. Please correct me if I am wrong with my analysis.